### PR TITLE
#17968 rename service integrations to Apps

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
Ticket  17968 says:
Most systems that offer integrations like this call it "Apps". We need to refactor our portlet, packaging and naming to call the new Portlet "Apps"

And So I did.. I renamed the whole thing to apps. I got rid of all the uses of the service-integration term.

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java [hunk 1]
```diff
@@ -141,44 +141,44 @@ public void Test_Create_service_descriptor_Then_Create_Service_Integration_Then_
 
         when(request.getRequestURI()).thenReturn("/baseURL");
 
-        final String serviceKey = String.format("lol_%d", System.currentTimeMillis());
-        final String fileName = String.format("%s.yml", serviceKey);
-        try(final InputStream inputStream = createServiceDescriptorFile(fileName, serviceKey, "lola",
+        final String appKey = String.format("lol_%d", System.currentTimeMillis());
+        final String fileName = String.format("%s.yml", appKey);
+        try(final InputStream inputStream = createAppDescriptorFile(fileName, appKey, "lola",
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java]: Codacy Issue found: [The String literal "lola" appears 9 times in this file; the first occurrence is on line 146](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5218098)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/AppsHelper.java [hunk 2]
```diff
@@ -136,75 +138,73 @@ Response getServiceIntegrationSiteView(
...
             final User user)
             throws DotSecurityException, DotDataException {
-        final Optional<ServiceDescriptor> serviceDescriptorOptional = serviceIntegrationAPI
-                .getServiceDescriptor(serviceKey, user);
-        if (serviceDescriptorOptional.isPresent()) {
-            final ServiceDescriptor serviceDescriptor = serviceDescriptorOptional.get();
+        final Optional<AppDescriptor> appDescriptorOptional = appsAPI
+                .getAppDescriptor(key, user);
+        if (appDescriptorOptional.isPresent()) {
+            final AppDescriptor appDescriptor = appDescriptorOptional.get();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/AppsHelper.java]: Codacy Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5218098)

File: dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java [hunk 3]
```diff
@@ -310,7 +311,7 @@ public void deleteSecrets(final String serviceKey, final Host host, final User u
     private static final String DESCRIPTORS_MAPPED_BY_SERVICE_KEY = "DESCRIPTORS_MAPPED_BY_SERVICE_KEY";
 
     @Override
-    public List<ServiceDescriptor> getServiceDescriptors(User user)
+    public List<AppDescriptor> getAppDescriptors(User user)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java]: Codacy Issue found: [Parameter 'user' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5218098)

File: dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java [hunk 4]
```diff
@@ -549,52 +550,52 @@ private static String getServiceDescriptorDirectory() {
...
-   private boolean validateServiceDescriptor(final ServiceDescriptor serviceDescriptor)
+   private boolean validateServiceDescriptor(final AppDescriptor appDescriptor)
            throws DotDataException {
-       if(UtilMethods.isNotSet(serviceDescriptor.getKey())){
+       if(UtilMethods.isNotSet(appDescriptor.getKey())){
           throw new DotDataException("The required field `key` isn't set on the incoming file.");
        }
 
-       if(serviceDescriptor.getKey().length() > 100){
+       if(appDescriptor.getKey().length() > 100){
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java]: Codacy Issue found: [Avoid using Literals in Conditional Statements](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5218098)

File: dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java [hunk 5]
```diff
@@ -67,23 +68,23 @@
...
         try {
-            return jsonMapper.readValue(json, ServiceSecrets.class);
+            return jsonMapper.readValue(json, AppSecrets.class);
         } catch (IOException e) {
             throw new DotDataException(e);
         }
     }
 
-    private ServiceSecrets readJson(final char[] chars) throws DotDataException {
+    private AppSecrets readJson(final char[] chars) throws DotDataException {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java]: Codacy Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5218098)

File: dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java [hunk 6]
```diff
@@ -42,25 +42,25 @@ public static void prepare() throws Exception {
     public void Test_Store_Json_Then_Recover_Decrypted_Then_Delete_Then_Verify_Its_Gone()
             throws DotDataException, DotSecurityException {
         final User admin = TestUserUtils.getAdminUser();
-        ServiceSecrets.Builder builder = new ServiceSecrets.Builder();
-        final String serviceKey = "anyServiceKey";
-        final ServiceSecrets bean = builder.withServiceKey(serviceKey)
+        AppSecrets.Builder builder = new AppSecrets.Builder();
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java]: Codacy Issue found: [Local variable 'builder' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5218098)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java [hunk 7]
```diff
@@ -624,34 +624,34 @@ public void Test_Pagination_And_Sort_Then_Request_Filter_Expect_Empty_Results()
         final HttpServletRequest request = mock(HttpServletRequest.class);
         final HttpServletResponse response = mock(HttpServletResponse.class);
         when(request.getRequestURI()).thenReturn("/baseURL");
-        final String serviceKey = String.format("lol_%d", System.currentTimeMillis());
-        final String fileName = String.format("%s.yml", serviceKey);
-        try (final InputStream inputStream = createServiceDescriptorFile(fileName, serviceKey,
+        final String key = String.format("lol_%d", System.currentTimeMillis());
+        final String fileName = String.format("%s.yml", key);
+        try (final InputStream inputStream = createAppDescriptorFile(fileName, key,
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java]: Codacy Issue found: [Unnecessary modifier 'final' on resource specification 'inputStream': resource specifications are implicitly final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5218098)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/AppsResource.java [hunk 8]
```diff
@@ -110,10 +110,10 @@ public final Response listAvailableServices(@Context final HttpServletRequest re
     @JSONP
     @NoCache
     @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
-    public final Response getServiceIntegrationByKey(
+    public final Response getAppByKey(
             @Context final HttpServletRequest request,
             @Context final HttpServletResponse response,
-            @PathParam("key") final String serviceKey,
+            @PathParam("key") final String key,
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/AppsResource.java]: Codacy Issue found: [The String literal "key" appears 4 times in this file; the first occurrence is on line 116](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5218098)

File: dotCMS/src/main/java/com/dotcms/security/apps/AppsAPI.java [hunk 9]
```diff
@@ -16,114 +16,116 @@
 import java.util.Set;
 
 /**
- * Entry Point to manage secrets and third party service integrations
+ * Entry Point to manage secrets and third party App integrations
  */
-public interface ServiceIntegrationAPI {
+public interface AppsAPI {
+
+     static final String APPS_API_IMPL = "APPS_API_IMPL";
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/security/apps/AppsAPI.java]: Codacy Issue found: [Unnecessary modifiers 'static final' on field 'APPS_API_IMPL': the field is declared in an interface type](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5218098)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java [hunk 10]
```diff
@@ -141,44 +141,44 @@ public void Test_Create_service_descriptor_Then_Create_Service_Integration_Then_
 
         when(request.getRequestURI()).thenReturn("/baseURL");
 
-        final String serviceKey = String.format("lol_%d", System.currentTimeMillis());
-        final String fileName = String.format("%s.yml", serviceKey);
-        try(final InputStream inputStream = createServiceDescriptorFile(fileName, serviceKey, "lola",
+        final String appKey = String.format("lol_%d", System.currentTimeMillis());
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java]: Codacy Issue found: [The String literal "lol_%d" appears 5 times in this file; the first occurrence is on line 144](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5218098)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java [hunk 11]
```diff
@@ -443,58 +443,58 @@ public void Test_Protected_Hidden_Secret() {
 
         when(request.getRequestURI()).thenReturn("/baseURL");
 
-        final String serviceKey = String.format("lol_%d", System.currentTimeMillis());
-        final String fileName = String.format("%s.yml", serviceKey);
-        try(final InputStream inputStream = createServiceDescriptorFile(fileName, serviceKey,
-                serviceKey,
+        final String appKey = String.format("lol_%d", System.currentTimeMillis());
+        final String fileName = String.format("%s.yml", appKey);
+        try(final InputStream inputStream = createAppDescriptorFile(fileName, appKey,
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java]: Codacy Issue found: [Unnecessary modifier 'final' on resource specification 'inputStream': resource specifications are implicitly final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5218098)

File: dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java [hunk 12]
```diff
@@ -91,122 +91,122 @@ public void Test_Store_Json_On_Different_Sites_Then_Pull_Sites_By_Host_Verify_Ma
         final Host host1 = new SiteDataGen().nextPersisted();
         final Host host2 = new SiteDataGen().nextPersisted();
 
-        final ServiceSecrets.Builder builder1 = new ServiceSecrets.Builder();
+        final AppSecrets.Builder builder1 = new AppSecrets.Builder();
 
-        final ServiceSecrets bean1Host1 = builder1.withServiceKey("serviceKey-1-Host-1")
+        final AppSecrets bean1Host1 = builder1.withKey("appKey-1-Host-1")
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java]: Codacy Issue found: [The String literal "appKey-1-Host-1" appears 10 times in this file; the first occurrence is on line 96](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5218098)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java [hunk 13]
```diff
@@ -248,48 +248,48 @@ public void Test_Create_service_descriptor_Then_Create_Service_Integration_Then_
         final HttpServletRequest request = mock(HttpServletRequest.class);
         final HttpServletResponse response = mock(HttpServletResponse.class);
         when(request.getRequestURI()).thenReturn("/baseURL");
-        final String serviceKey = String.format("lol_%d", System.currentTimeMillis());
-        final String fileName = String.format("%s.yml", serviceKey);
-        try(final InputStream inputStream = createServiceDescriptorFile(fileName, serviceKey, "lola",
+        final String appKey = String.format("lol_%d", System.currentTimeMillis());
+        final String fileName = String.format("%s.yml", appKey);
+        try(final InputStream inputStream = createAppDescriptorFile(fileName, appKey, "lola",
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java]: Codacy Issue found: [Unnecessary modifier 'final' on resource specification 'inputStream': resource specifications are implicitly final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5218098)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java [hunk 14]
```diff
@@ -522,42 +522,42 @@ public void Test_Service_Key_Casing() throws IOException, DotDataException, DotS
 
         long time = System.currentTimeMillis();
 
-        final String serviceKey1 = String.format("all_lower_case_not_too_short_prefix_%d", time);
-        final String fileName = String.format("%s.yml", serviceKey1);
-        try(final InputStream inputStream = createServiceDescriptorFile(fileName, serviceKey1,
-                serviceKey1,
+        final String appKey1 = String.format("all_lower_case_not_too_short_prefix_%d", time);
+        final String fileName = String.format("%s.yml", appKey1);
+        try(final InputStream inputStream = createAppDescriptorFile(fileName, appKey1,
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java]: Codacy Issue found: [Unnecessary modifier 'final' on resource specification 'inputStream': resource specifications are implicitly final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5218098)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java [hunk 15]
```diff
@@ -141,44 +141,44 @@ public void Test_Create_service_descriptor_Then_Create_Service_Integration_Then_
 
         when(request.getRequestURI()).thenReturn("/baseURL");
 
-        final String serviceKey = String.format("lol_%d", System.currentTimeMillis());
-        final String fileName = String.format("%s.yml", serviceKey);
-        try(final InputStream inputStream = createServiceDescriptorFile(fileName, serviceKey, "lola",
+        final String appKey = String.format("lol_%d", System.currentTimeMillis());
+        final String fileName = String.format("%s.yml", appKey);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java]: Codacy Issue found: [The String literal "%s.yml" appears 7 times in this file; the first occurrence is on line 145](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5218098)

File: dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java [hunk 16]
```diff
@@ -242,13 +242,13 @@ public void Test_Save_Secrets_Then_Update_Single_Secret_Verify_Updated_Value() t
...
-        final ServiceIntegrationAPI api = APILocator.getServiceIntegrationAPI();
+        final AppsAPI api = APILocator.getAppsAPI();
         final User admin = TestUserUtils.getAdminUser();
         final Host host = new SiteDataGen().nextPersisted();
 
         //Let's create a set of secrets for a service
-        final ServiceSecrets.Builder builder1 = new ServiceSecrets.Builder();
-        final ServiceSecrets secrets1 = builder1.withServiceKey("serviceKeyHost-1")
+        final AppSecrets.Builder builder1 = new AppSecrets.Builder();
+        final AppSecrets secrets1 = builder1.withKey("appKeyHost-1")
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java]: Codacy Issue found: [The String literal "appKeyHost-1" appears 7 times in this file; the first occurrence is on line 251](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5218098)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/AppsHelper.java [hunk 17]
```diff
@@ -380,50 +380,50 @@ private void validateIncomingParams(final Set<String> incomingParamNames, final
...
      */
-    List<ServiceIntegrationView> createServiceIntegration(final FormDataMultiPart multipart, final User user)
+    List<AppView> createApp(final FormDataMultiPart multipart, final User user)
             throws IOException, DotDataException {
         final List<File> files = new MultiPartUtils().getBinariesFromMultipart(multipart);
         if(!UtilMethods.isSet(files)){
             throw new DotDataException("Unable to extract any files from multi-part request.");
         }
-        List<ServiceIntegrationView> serviceIntegrationViews = new ArrayList<>(files.size());
+        List<AppView> appViews = new ArrayList<>(files.size());
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/AppsHelper.java]: Codacy Issue found: [Local variable 'appViews' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5218098)

**Review by jdotcms — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Pull request merged**